feat: create ConfirmTitle component

digiwand opened the pull request:
## **Description**

New component to display the confirmation title and subtitle

Notes:
- storybook page created
- I put the custom hooks directly in the confirm/title directory because it is not being used outside of this directory

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/21282
Blocked By: https://github.com/MetaMask/metamask-extension/pull/21970

## **Manual testing steps**

## **Screenshots/Recordings**

see blue box in below image:
<img width="693" alt="confirmtitle" src="https://github.com/MetaMask/metamask-extension/assets/20778143/4582858f-9574-40a9-aee4-cbcfc7c517f9">

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

digiwand committed:
feat: create ConfirmTitle
todo: update to use hooks

digiwand committed:
refactor: rn ConfirmTitle subtitle -> description

digiwand committed:
feat: ConfirmTitle use hooks

digiwand committed:
feat: ConfirmTitle use memo

digiwand committed:
refactor: rm hooks for ConfirmTitle

digiwand committed:
fix:lint: alphabetize

digiwand committed:
refactor: reverse selector from ef72699

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand committed:
refactor: update type Confirmation; fix eslint

jpuri reviewed (dismissed):
👍

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand commented on ui/components/app/confirm/title/title.tsx:
```diff
@@ -0,0 +1,51 @@
+
+const typeToDescTKey: Partial<Record<TransactionType, string>> = {
+  [TransactionType.personalSign]: 'confirmTitleDescSignature',
+};
```
essentially, would we be using 

current:
```
<ConfirmTitle> 
```
or

passed values:
```
<ConfirmTitle title={} description={}>
```
?

segun replied:
I think @jpuri mentioned we want all the components to handle their logic and not pass props.

digiwand replied:
great mention @segun. I also thought this, though, for our ConfirmInfo component, @jpuri requested to pass props.

double-checking here

jpuri replied:
Hey @digiwand : sorry for delayed response, this code looks good. I think we do not need to pass values.

georgewrmarshall commented on app/_locales/en/messages.json:
```diff
@@ -792,6 +792,12 @@
     "message": "Confirm Secret Recovery Phrase"
   },
+  "confirmTitleDescSignature": {
+    "message": "Only sign this message if you fully understand the content and trust the requesting site."
```
I believe single line sentences should not contain a period. cc @coreyjanssen https://www.notion.so/MetaMask-Content-Design-Style-Guide-7d0b7579786540afbe02780ebb203254?pvs=4#7ffcee3343bc4115991c7a7e0d2f1c71

![Screenshot 2024-01-09 at 4 41 22 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/9c270f17-3355-4350-ab1c-346845d7488c)

digiwand replied:
lgtm! updated d6f448e191195ee96d29050f45c7c8e9272d55c4

thanks @georgewrmarshall

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand committed:
fix: ConfirmTitle rm period in copy
https://github.com/MetaMask/metamask-extension/pull/22270/files#r1446766872

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

georgewrmarshall reviewed (dismissed):
Content LGTM!

digiwand committed:
fix:test: rm "." from single line text

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

segun reviewed (changes requested):
Would be nice to add storybook as well.

digiwand committed:
refactor: update ConfirmTitle import/export

digiwand committed:
feat: create ConfirmTitle storybook page

digiwand commented:
hi @segun, thanks for the review!

I was holding off on creating storybook pages for this reason:

> no storybook page created for this. We can display this within the confirm page storybook page(s) when we create them. see: https://storybook.js.org/docs/writing-stories/build-pages-with-storybook#args-composition-for-presentational-screens

however, I've added it now upon request since it could be useful. It's a simple storybook page for now, so it doesn't have controls yet we don't support other variants at this time

updated [5983f88](https://github.com/MetaMask/metamask-extension/pull/22270/commits/5983f8836aca54dff199439fe828de86e164a467)

georgewrmarshall commented:
> hi @segun, thanks for the review!
> 
> I was holding off on creating storybook pages for this reason:
> 
> > no storybook page created for this. We can display this within the confirm page storybook page(s) when we create them. see: https://storybook.js.org/docs/writing-stories/build-pages-with-storybook#args-composition-for-presentational-screens
> 
> however, I've added it now upon request since it could be useful. It's a simple storybook page for now, so it doesn't have controls yet we don't support other variants at this time
> 
> updated [5983f88](https://github.com/MetaMask/metamask-extension/pull/22270/commits/5983f8836aca54dff199439fe828de86e164a467)

I also think a storybook file no matter how simple the component is can be helpful :)

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand commented:
gotchya, thanks for the direction on storybook pages here @segun & @georgewrmarshall!

digiwand committed:
fix: import default ConfirmTitle

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2

digiwand commented:
![Screenshot 2024-01-29 at 4 41 00 PM](https://github.com/MetaMask/metamask-extension/assets/20778143/e9867e8f-8fa7-4cab-95fc-9a939740cc50)

hi @segun,
not seeing a way to dismiss or mark the change request as complete. may I get a rereview please?

digiwand committed:
Merge branch 'develop' into feat-confirm-title-with-description-2
